File: app/src/main/java/com/constraint/vagabond/main/search/adapter/AreaAdapterPresenter.java
package com.constraint.vagabond.main.search.adapter;

import com.constraint.vagabond.data.entities.RecAreaMedia;
import com.constraint.vagabond.data.entities.RecreationalArea;
import com.constraint.vagabond.data.entities.RecreationalAreaList;

import java.util.List;

import androidx.annotation.NonNull;

public class AreaAdapterPresenter {

  private final RecreationalAreaList recreationalAreaList;

  public AreaAdapterPresenter(RecreationalAreaList recreationalAreaList) {
    this.recreationalAreaList = recreationalAreaList;
  }

  public void onBindDataToAdapter(@NonNull AreaHolder holder, int position) {
    final RecreationalArea recreationalArea =
        recreationalAreaList.getRecreationalAreaList().get(position);
    holder.setAreaTitle(recreationalArea.getRecAreaName());
    final List<RecAreaMedia> recAreaMedia = recreationalArea.getRecAreaMediaList();
    holder.setAreaImage(recAreaMedia.get(0).getImageURL());
  }

  public int getAreaCount() {
    return recreationalAreaList.getRecreationalAreaList().size();
  }
}

File: app/src/main/java/com/constraint/vagabond/main/search/adapter/AreaAdapterPresenter.kt
package com.constraint.vagabond.main.search.adapter

import com.constraint.vagabond.data.entities.RecreationalAreaList

class AreaAdapterPresenter internal constructor(private val recreationalAreaList: RecreationalAreaList) {

    internal val areaCount: Int
        get() = recreationalAreaList.recreationalAreaList.size

    internal fun onBindDataToAdapter(holder: AreaHolder, position: Int) {
        val (_, recAreaName, _, _, _, _, recAreaMedia) = recreationalAreaList.recreationalAreaList[position]
        holder.setAreaTitle(recAreaName)
        if (recAreaMedia.isEmpty()) {
            holder.setAreaImage(null)
        } else {
            holder.setAreaImage(recAreaMedia[0].imageURL)
        }

    }
}
